elementary/monitor/api/filters/filters.py

```diff
@@ -50,7 +51,8 @@ def _get_test_filters(
 
         totals_models_ids = totals.keys()
         for artifact in [*models.values(), *sources.values()]:
-            if artifact.unique_id not in totals_models_ids:
+            assert isinstance(artifact, ArtifactSchema)
```

Comment:
I added asserts in many places where I needed to [narrow down the type](https://mypy.readthedocs.io/en/stable/type_narrowing.html) for mypy

I think in this case it's fine to use asserts as I'm not relying on them as production checks

Reply:
Wouldn't it be preferable to use [`cast()`](https://mypy.readthedocs.io/en/stable/type_narrowing.html#casts) in that case?